src/components/StrongPasswordInput/index.js

```diff
@@ -0,0 +1,248 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { uniqueId } from '../../libs/utils';
+import Label from '../Input/label';
+import StyledIconContainer from '../Input/styled/iconContainer';
+import RelativeElement from '../Structural/relativeElement';
+import StyledInput from './styled/input'; // TODO: change to import from index
+import RenderIf from '../RenderIf';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import StrengthBar from './strengthBar';
+import { StyledContainer } from './styled';
+
+/**
+ * StrongPasswordInput component is used to provide feedback about password strength
+ * @category Form
+ */
+const StrongPasswordInput = React.forwardRef((props, ref) => {
+    const inputId = uniqueId('input');
+    const errorMessageId = uniqueId('error-message');
+    const inlineTextLabelId = uniqueId('inline-text-label');
+
+    function getInlineTextLabelId() {
+        const { bottomHelpText } = props;
+        if (bottomHelpText) {
+            return inlineTextLabelId;
+        }
+        return undefined;
+    }
+
+    function getErrorMessageId() {
+        const { error } = props;
+        if (error) {
+            return errorMessageId;
+        }
+        return undefined;
+    }
```

Comment:
This logic is repetitive for various components. you can do 2 global custom hook to be able to reuse them in other components. put the hooks in the src / lib / hooks folder